src/sprintf.js

```diff
@@ -131,7 +131,7 @@
                     }
                     pad_character = ph.pad_char ? ph.pad_char === '0' ? '0' : ph.pad_char.charAt(1) : ' '
                     pad_length = ph.width - (sign + arg).length
-                    pad = ph.width ? (pad_length > 0 ? pad_character.repeat(pad_length) : '') : ''
+                    pad = ph.width ? (pad_length > 0 ? new Array(pad_length + 1).join(pad_character) : '') : ''
```

Comment:
This _might_ be concern of optimization - native String.repeat I _assume_ is not only faster but also simpler.

It could also do a feature check:

`pad = ph.width ? (pad_length > 0 ? String.prototype.hasOwnProperty('repeat') ? pad_character.repeat(pad_length) : new Array(pad_length + 1).join(pad_character) : '') : ''`